### PR TITLE
Propose mirgration of codebase from JS to TS

### PR DESCRIPTION
**Pull request by theofilis**
Hi,

Project Iroha is very interesting thus I want to contribute my coding skill in yours javascript library.

I migrate your project to Typescript I use the latest trends on library development with Typescript as they are proposed [here](https://github.com/synaphea/typescript-library-starter)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: node_js
```
Comment (takemiyamakoto) [inline .travis.yml]: Why travis? Other Iroha projects are using circle-ci2.

File: package.json [hunk 2]
```diff
@@ -2,27 +2,111 @@
...
+  "typings": "dist/src/irohajs.d.ts",
+  "files": [
+    "dist"
+  ],
+  "author": "",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/soramitsu/iroha-javascript/issues"
+  },
+  "homepage": "https://github.com/soramitsu/iroha-javascript#readme",
```
Comment (takemiyamakoto) [inline package.json]: ```https://github.com/hyperledger/iroha-javascript#readme``` might be better

File: package.json [hunk 3]
```diff
@@ -2,27 +2,111 @@
...
+  "keywords": [],
+  "main": "dist/irohajs.js",
+  "typings": "dist/src/irohajs.d.ts",
+  "files": [
+    "dist"
+  ],
+  "author": "",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/soramitsu/iroha-javascript/issues"
```
Comment (takemiyamakoto) [inline package.json]: ```https://github.com/hyperledger/iroha-javascript/issues``` might be better

File: package.json [hunk 4]
```diff
@@ -2,27 +2,111 @@
...
+    "deploy-docs": "node tools/gh-pages-publish.js",
+    "report-coverage": "cat ./coverage/lcov.info | coveralls",
+    "commit": "git-cz",
+    "semantic-release": "semantic-release pre && npm publish && semantic-release post",
+    "prepush": "npm run test:prod && npm run build",
+    "commitmsg": "validate-commit-msg"
   },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/soramitsu/iroha-javascript.git"
```
Comment (takemiyamakoto) [inline package.json]: →``` "url": "git+https://github.com/hyperledger/iroha-javascript.git"```

**Comment by takemiyamakoto** [comment]
Overall, this looks okay. I guess there is just the question on whether we want to switch to typescript from JS. I think more people than me should comment on this, so let's see what people think.

File: .travis.yml (hunk 1 above)
Comment (theofilis) [inline .travis.yml]: Sorry, i didn't know you already use circle-ci2. I would try to change it to be like other one.

File: package.json (hunk 3 above)
Comment (theofilis) [inline package.json]: οκ

**Comment by theofilis** [comment]
In this article you could find reasons why it's good idea to adopt Typescript

https://medium.freecodecamp.com/when-should-i-use-typescript-311cb5fe801b#.xxvrfzoa6

**Comment by takemiyamakoto** [comment]
I personally like Typescript. I think we can have a decision today.

**Pull request merged**